### PR TITLE
libnx port

### PR DESCRIPTION
**Pull request by billy-acuna**
@m4xw @natinusala

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -204,6 +204,21 @@ else ifeq ($(platform), switch)
...
 	fpic := -nostdlib
 
+# Nintendo Switch (libnx)
+else ifeq ($(platform), libnx)
+    include $(DEVKITPRO)/libnx/switch_rules
+    EXT=a
+    TARGET := $(TARGET_NAME)_libretro_$(platform).$(EXT)
+    DEFINES := -DSWITCH=1 -U__linux__ -U__linux -DRARCH_INTERNAL
+    CFLAGS  :=  $(DEFINES) -g \
+                -O2 \
```
Comment (m4xw) [inline Makefile]: Please use O3 with -ftree-vectorize
Also remove the "\" from the lone O2
This statement will need to be changed too https://github.com/libretro/virtualjaguar-libretro/blob/bb43634b549e7a27ab92a6eebeaa9cb842222e64/Makefile#L526

Comment (m4xw) [reply Makefile]: With statement I mean the if around that

**Comment by billy-acuna** [comment]
Done @m4xw :shrek:

**Comment by m4xw** [comment]
@billy-acuna You broke like everything

**Pull request closed without merging**